fvm/evm/evm.go

```diff
@@ -53,15 +57,30 @@ func SetupEnvironment(
 
 	addressAllocator := handler.NewAddressAllocator()
 
+	height, err := backend.GetCurrentBlockHeight()
+	if err != nil {
+		return err
+	}
+
+	block, ok, err := backend.GetBlockAtHeight(height)
+	if err != nil {
+		return err
+	}
+	if !ok {
+		return fmt.Errorf("could not retrieve the block at height")
+	}
```

Comment:
any chance we could do this only if the tracer is set?

Reply:
discussing with @ramtinms he suggested we do it always